docs/hooks/overview.md

```diff
@@ -67,7 +67,9 @@ This allows configuration-as-code inside lakeFS, where Action files are declarat
 {: .no_toc }
 
 An **Action** is a list of Hooks with the same trigger configuration, i.e. an event will trigger all Hooks under an Action or none at all.
-The Hooks under an Action are ordered and so is their execution. A Hook will only be executed if all the previous Hooks that were triggered with it had passed.
+The Hooks under an Action are ordered and so is their execution.
+Before each hook execution the `if` boolean expression is evaluated. The expression can use `success()` which evaluate to true in case of no error was occured while running the hoook's actions. It can also use `failure()` which evaluate to true when previous step failed - can be used to report failures.
+By default, when `if` is empty or omitted, the step will run only if no error occurred (same as evaluate of success function).
```

Comment:
very similar to GitHub actions. it evaluates to a check of the last error that is set once a hook fails. It is what we did so far, just broke the processing loop. now we continue to evaluate each step.

Reply:
Yes but how would you write the condition to continue in case of errors in practice? I think this might be confusing

Reply:
Maybe I don't understand the question. But in practice you write `if: success()` or nothing in case you like the step to execute if everything green until that point.
In case you like to run always `if: true`.
In case you like to run if something failed `if: failure()`.